src/hotspot/share/memory/memRegion.hpp

```diff
@@ -63,7 +63,8 @@ class MemRegion {
   MemRegion minus(const MemRegion mr2) const;
 
   HeapWord* start() const { return _start; }
-  HeapWord* end() const   { return _start + _word_size; }
+  // in the gtests we call end() with a _start == nullptr so adjust the addition to avoid ub
+  HeapWord* end() const   { return reinterpret_cast<HeapWord*>(reinterpret_cast<uintptr_t>(_start) + (_word_size * sizeof(HeapWord))); }
```

Comment:
I don't think this change should be made. Instead, fix the offending test.  The fake heap it creates starts
at null, but I think it could be anywhere, because nothing touches it.

Reply:
If it did (or does) need to be valid memory, then just allocate a chunk of
native memory for this fake heap for the test. That might be the safer thing
to do anyway, in case some future change leads to the fake region getting
touched.

Reply:
> I don't think this change should be made. Instead, fix the offending test. The fake heap it creates starts at null, but I think it could be anywhere, because nothing touches it.

Let's do it this way !